src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +395,468 @@ else if (!period.isEmpty()) {
+        DetailNode node = null;
+        final DetailNode[] children = javadoc.getChildren();
+        for (DetailNode child: children) {
+            // If present as a children of javadoc
+            if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                node = child;
+            }
+            // If nested inside html tag
+            else if (child.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+                node = getInlineTagNodeWithinHtmlElement(child);
+            }
+
+            if (node != null) {
+                break;
+            }
+        }
+        return node;
+    }
+
+    /**
+     * Returns inline javadoc tag node inside html tags from a HTML tag.
+     *
+     * @param ast html tag node.
+     * @return Inline summary javadoc tag node or null if no node is found.
+     */
+    private static DetailNode getInlineTagNodeWithinHtmlElement(DetailNode ast) {
+        DetailNode node = ast;
+        DetailNode result = null;
+        // node can never be null as it is always called when there is HTML element
+        if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+            result = node;
+        }
+        else if (node.getType() == JavadocTokenTypes.HTML_TAG) {
+            // HTML_TAG always has more than 2 children.
+            node = node.getChildren()[1];
+            result = getInlineTagNodeWithinHtmlElement(node);
+        }
+        else if (node.getType() == JavadocTokenTypes.HTML_ELEMENT
+                // Condition for SINGLETON html element which cannot contain summary node
+                && node.getChildren()[0].getChildren().length > 1) {
+            // Html elements have one tested tag before actual content inside it
+            node = node.getChildren()[0].getChildren()[1];
+            result = getInlineTagNodeWithinHtmlElement(node);
+        }
+        return result;
+    }
+
+    /**
+     * Checks if the first tag inside ast is summary tag.
+     *
+     * @param javadoc root node.
+     * @return true, if first tag is summary tag.
+     */
+    private static boolean isSummaryTag(DetailNode javadoc) {
+        final DetailNode[] child = javadoc.getChildren();
+
+        // Checking size of ast is not required, since ast contains
+        // children of Inline Tag, as at least 2 children will be present which are
+        // RCURLY and LCURLY.
+        return child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                && SUMMARY_TEXT.equals(child[1].getText());
+    }
+
+    /**
+     * Check the summary of inline form.
+     *
+     * @param ast Javadoc root Node.
+     */
+    private void validateSummaryTag(DetailNode ast) {
+        final String inlineSummary = getInlineSummary(ast);
+        if (inlineSummary.isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            final String inlineFirstSentence = getInlineFirstSentence(ast);
```

Comment:
please read attentively https://github.com/checkstyle/checkstyle/issues/7202#issuecomment-766890211
we do not care about content of javadoc tag, all we need is it is last HTML printable symbol is `.`. to make looks good in HTML table with summary content.

all test pass on this update:
```
✔ ~/java/github/romani/checkstyle [AkMo3/FixNewSummaryFormat L|✚ 2] 
15:17 $ git diff
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
index 4b8645e..5b5f920 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
@@ -487,8 +487,7 @@ public class SummaryJavadocCheck extends AbstractJavadocCheck {
             log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
         }
         else if (!period.isEmpty()) {
-            final String inlineFirstSentence = getInlineFirstSentence(ast);
-            final boolean isPeriodAbsentInFirstSentence = !inlineFirstSentence.contains(period);
+            final boolean isPeriodAbsentInFirstSentence = !inlineSummary.contains(period);
             if (isPeriodAbsentInFirstSentence) {
                 log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
             }
diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java
index ca5c0b3..88523d5 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheckTest.java
@@ -105,7 +105,7 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
             "63: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "74: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "79: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
-            "89: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
+            "89: " + getCheckMessage(MSG_SUMMARY_JAVADOC),
             "103: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "122: " + getCheckMessage(MSG_SUMMARY_JAVADOC),
             "128: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
@@ -176,9 +176,7 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
             "54: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "72: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "77: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
-            "87: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
             "93: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
-            "98: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
             "109: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "144: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
             "149: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
@@ -193,11 +191,8 @@ public class SummaryJavadocCheckTest extends AbstractModuleTestSupport {
             "203: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "208: " + getCheckMessage(MSG_SUMMARY_JAVADOC_MISSING),
             "216: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "228: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
             "240: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
             "253: " + getCheckMessage(MSG_SUMMARY_MISSING_PERIOD),
-            "265: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
-            "278: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
         };
 
         createChecker(checkConfig);
```

Reply:
Sir, we do need to care about the content inside the inline tag as discussed in [#9174 comment](https://github.com/checkstyle/checkstyle/pull/9174#discussion_r598729620), there are three possible cases of violation and for all of them we need to check the content inside the tag and those are

 1. No `PERIOD` after the entire summary
 2. No `PERIOD` after first sentence of summary.
 3. Invalid tags inside Javadoc tag.
 
> we do not care about content of javadoc tag, all we need is it is last HTML printable symbol is .. to make looks good in HTML table with summary content.

By this, only the first condition will be checked, while there are others too.

Reply:
>No PERIOD after first sentence of summary.

First sentence in old javadoc was as summary.
Such summary was placed in special table in html. It was a problem so special tag was introduced. So multiple sentences moved summary table. So in tag first sentence is not required, whole content of tag is content for summary table so whole tag content should ends with period

Reply:
> Lets create new message for Check for **this case** Summary of Javadoc is missing an ending period.

_**this case**_ here has caused confusion that we need to separately check for both cases. Fix done.